DotNetHowToMultipleDataSources/Program.cs

```diff
@@ -0,0 +1,153 @@
+            // We can clear the index by deleting it if it exists and creating
+            // it again
+            bool exists = await searchService.Indexes.ExistsAsync(index.Name);
+            if (exists)
+            {
+                await searchService.Indexes.DeleteAsync(index.Name);
+            }
+
+            await searchService.Indexes.CreateAsync(index);
+
+            Console.WriteLine("Creating data sources...");
+
+            // The SQL sample data set has a table name of "hotels"
+            // The SQL sample data set uses SQL integrated change tracking for change detection
+            // This means that when the SQL indexer runs, it will be able to detect which data has
+            // changed since the last run using built in change tracking
+            // See this link for more information
+            // https://docs.microsoft.com/sql/relational-databases/track-changes/about-change-tracking-sql-server
+            DataSource sqlDataSource = DataSource.AzureSql(
+                name: "azure-sql",
+                sqlConnectionString: configuration["AzureSQLConnectionString"],
+                tableOrViewName: "hotels");
+            sqlDataSource.DataChangeDetectionPolicy = new SqlIntegratedChangeTrackingPolicy();
+            // The SQL data source does not need to be deleted if it was already created,
+            // but the connection string may need to be updated if it was changed
+            await searchService.DataSources.CreateOrUpdateAsync(sqlDataSource);
+
+            // The JSON sample data set has a collection name of "hotels"
+            // The JSON sample data set uses Cosmos DB change tracking for change detection
+            // This means that when the Cosmos DB indexer runs, it will be able to detect which data has
+            // changed since the last run using built in change tracking
+            // See this link for more information
+            // https://docs.microsoft.com/azure/search/search-howto-index-cosmosdb#indexing-changed-documents
+            DataSource cosmosDbDataSource = DataSource.DocumentDb(
+                name: "cosmos-db",
+                documentDbConnectionString: configuration["CosmosDBConnectionString"],
+                collectionName: "hotels",
+                useChangeDetection: true);
+            // The Cosmos DB data source does not need to be deleted if it was already created,
+            // but the connection string may need to be updated if it was changed
+            await searchService.DataSources.CreateOrUpdateAsync(cosmosDbDataSource);
+
+            Console.WriteLine("Creating Azure SQL indexer...");
+            Indexer sqlIndexer = new Indexer(
+                name: "azure-sql-indexer",
+                dataSourceName: sqlDataSource.Name,
+                targetIndexName: index.Name,
+                schedule: new IndexingSchedule(TimeSpan.FromDays(1)));
+            // Indexers contain metadata about how much they have already indexed
```

Comment:
@mgottein You could avoid the need for this by just deleting and re-creating the indexer. Arguably this sample should clean up after itself anyway.

Reply:
The sample should poll for a successful indexer, then delete everything?

Reply:
Thinking about it some more, I guess it depends on what you're trying to demonstrate. I've always used indexers in samples just as a shortcut to getting the data I need into a test index. But if your goal is to show best practices for managing indexers, maybe you want this code to act like the indexer is a long-living thing. Either way, it would help to clarify your intent with comments.

Reply:
If comments would help here, they'd help in the other samples too